### DISCUSSION TITLE
Protection against bots

### DISCUSSION THREAD

**Discussion by perkan777** [Web3Auth/Web3Auth#670] [Feature Requests]
Is web3auth in any way handling automated bot registrations?
We have a fear that our product might be abused by registrations and we don't see any "recaptcha" or other tools to handle those kind of situations.

**Comment by shahbaz17** [comment]
Hey @perkan777 

What's your stack, and which login options do you have for your users?

**Comment by yashovardhan** [comment]
You can set `mfaLevel` as `mandatory` in your app. That way one has to go through the whole registration process properly while being totally secure. I don't think bots will be able to pass this flow.

Check it out here: https://web3auth.io/docs/sdk/web/web3auth/multi-factor-authentication
